src/core/packet_builder.c

```diff
@@ -243,17 +243,24 @@ QuicPacketBuilderPrepare(
                 Builder->MinimumDatagramLength = NewDatagramLength;
             }
 
-        } else if (NewPacketType == QUIC_INITIAL &&
-            !QuicConnIsServer(Connection)) {
+        } else if (NewPacketType == QUIC_INITIAL) {
 
             //
-            // Make sure to pad the packet if client Initial packets.
+            // Make sure to pad Initial packets.
             //
             Builder->MinimumDatagramLength =
                 MaxUdpPayloadSizeForFamily(
                     QuicAddrGetFamily(&Builder->Path->RemoteAddress),
                     QUIC_INITIAL_PACKET_LENGTH);
 
+            if ((uint32_t)Builder->MinimumDatagramLength > Builder->Datagram->Length) {
```

Comment:
Doesn't `Builder->Datagram->Length` refer to the length of the buffer?

Reply:
The allocated length of the datagram.